### PR TITLE
Fix wrong casting to long

### PR DESCRIPTION
**Pull request by julia-zack**
- Fix wrong casting to long
- Change proposed federation creationTime to seconds instead of milliseconds
- Improve `svpIsOngoing` readability

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 1]
```diff
@@ -1019,13 +1019,17 @@ private void logUpdateCollections(Transaction rskTx) {
...
-            .filter(validationPeriodEndBlock -> rskExecutionBlock.getNumber() <= validationPeriodEndBlock)
+            .filter(this::validationPeriodIsOngoing)
             .isPresent();
     }
 
+    private boolean validationPeriodIsOngoing(Federation proposedFederation) {
+        long validationPeriodEndBlock = proposedFederation.getCreationBlockNumber() +
+            bridgeConstants.getFederationConstants().getValidationPeriodDurationInBlocks();
+
+        return rskExecutionBlock.getNumber() < validationPeriodEndBlock;
```
Comment (julia-zack) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:1030]: also replaced the `<=` for a `<` to follow the same logic as in migration period (i.e. not considering the ending block as part of the period)

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java [hunk 2]
```diff
@@ -912,7 +912,7 @@ private void recreateSvpSpendTransaction() {
                 .setScriptSig(createBaseP2SHInputScriptThatSpendsFromRedeemScript(flyoverRedeemScript));
 
             // add output
-            svpSpendTx.addOutput(Coin.valueOf(1762), federationSupport.getActiveFederationAddress());
+            svpSpendTx.addOutput(Coin.valueOf(2114L), federationSupport.getActiveFederationAddress());
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java:915]: `2114 = 1762 * 1.2` :)

File: rskj-core/src/main/java/co/rsk/peg/Bridge.java [hunk 3]
```diff
@@ -1085,22 +1085,22 @@ public int getProposedFederationSize(Object[] args) {
...
-     * @return the creation time of the proposed federation in milliseconds since the epoch,
+     * @return the creation time of the proposed federation in seconds since the epoch,
      *         or -1 if no proposed federation exists.
      */
     public Long getProposedFederationCreationTime(Object[] args) {
         logger.trace("getProposedFederationCreationTime");
 
         return bridgeSupport.getProposedFederationCreationTime()
-            .map(Instant::toEpochMilli)
+            .map(Instant::getEpochSecond)
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/Bridge.java:1103]: Now I wonder if we should also apply this change to active and retiring feds

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 4]
```diff
@@ -1119,10 +1123,10 @@ private void addSvpSpendTransactionInputs(BtcTransaction svpSpendTransaction, Bt
 
     private Coin calculateSvpSpendTxAmount(Federation proposedFederation) {
         int svpSpendTransactionSize = calculatePegoutTxSize(activations, proposedFederation, 2, 1);
-        long backupSizePercentage = (long) 1.2; // just to be sure the amount sent will be enough
+        long svpSpendTransactionBackedUpSize = svpSpendTransactionSize * 12L / 10L; // just to be sure the amount sent will be enough
 
         return feePerKbSupport.getFeePerKb()
-            .multiply(svpSpendTransactionSize * backupSizePercentage)
+            .multiply(svpSpendTransactionBackedUpSize)
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:1129]:
```suggestion
            .multiply(svpSpendTransactionBackedUpSize)
            .multiply(1200) // Add 20% just to be sure the amount sent will be enough
            .divide(1000)
```
Just an idea, maybe we encapsulate the increase here?

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java [hunk 5]
```diff
@@ -567,7 +567,7 @@ void processSvpSpendTransaction_createsExpectedTransactionAndSavesTheValuesAndLo
             assertSvpFundTxSignedWasRemovedFromStorage();
 
             assertLogPegoutTransactionCreated(logs, svpSpendTransactionUnsigned);
-            Coin valueSentToActiveFed = Coin.valueOf(1762);
+            Coin valueSentToActiveFed = Coin.valueOf(2114L);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java:570]: Where is this magic number coming from?

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java (hunk 2 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java:915]: And where did 1762 come from?

File: rskj-core/src/main/java/co/rsk/peg/Bridge.java (hunk 3 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/Bridge.java:1103]: I thought about that. I can create a task in the backlog so we dont forget

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 4 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:1129]: i found the other way clearer. We would end up having two `.divide(1000)` in a row, kind of confusing. Wdyt?

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java]: updated

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java (hunk 5 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java]: updated

**Pull request merged**